src/main/scala/de/zalando/beard/renderer/ForContextFactory.scala

```diff
@@ -23,7 +23,11 @@ object ForContextFactory {
           updated("isFirst", index == 0).
           updated("isOdd", index % 2 == 1).
           updated("isEven", index % 2 == 0)
-        newContext.updated(forIterationContext.templateIteratorIdentifier, newMap)
+
+        (forIterationContext.templateIndexIdentifier match {
```

Comment:
You should extract this to a new function with a proper name and write some unit test for it